### PR TITLE
feat(): delete contacted person

### PR DESCRIPTION
**Pull request by yaniryos**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx [hunk 1]
```diff
@@ -120,108 +122,111 @@ const InteractionEventForm: React.FC<Props> = (
...
+          </Grid>
+        </Grid>
+        <Divider light={true} />
+        <Grid container className={formClasses.form + ' ' + classes.spacedOutForm}>
+          {
+            contacts.map((contact, index: number) => (
+              <Grid item className={classes.contactFormItem} key={contact.serialId === -1 ? contact.id : contact.serialId}>
+                <ContactForm
+                  updatedContactIndex={index}
+                  onDeleteClick={() => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx:202]: maybe extract it to a function

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts [hunk 2]
```diff
@@ -68,7 +75,11 @@ const useInteractionsForm = (props : useInteractionFormIncome): useInteractionFo
         axios.post('/intersections/createContactEvent', {
           ...interactionsDataToSave,
           [InteractionEventDialogFields.LOCATION_ADDRESS]: locationAddress,
-          [InteractionEventDialogFields.INVESTIGATION_ID]: epidemiologyNumber
+          [InteractionEventDialogFields.INVESTIGATION_ID]: epidemiologyNumber,
+          contacts: interactionsDataToSave.contacts.map((contact: Contact) => ({
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:79]: why don't you use interactionEventDialogFields to get 'contacts'?

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 3]
```diff
@@ -183,6 +184,26 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
         });
     }
 
+    const handleDeleteEventFailed = () => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:187]: Maybe combine both functions and just pass message as a parameter

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 4]
```diff
@@ -233,24 +254,80 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
+                        step: `interaction was deleted successfully`,
+                        user: userId,
+                        investigation: epidemiologyNumber
+                    });
+
+                    let currentEvent = interactions.filter((interaction: InteractionEventDialogData) => interaction.id === contactEventId)[0];
+                    if (currentEvent) {
+                        currentEvent.contacts = currentEvent.contacts.filter((contact: Contact) => contact.serialId !== contactedPersonId)
+                    }
+                    setInteractions(interactions.map((interaction: InteractionEventDialogData) => (
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:298]: I think you can use loadInteractions instead

File: client/src/models/Contact.ts [hunk 5]
```diff
@@ -1,9 +1,9 @@
 interface Contact {
-    serialId?: number;
+    serialId: number;
```
Comment (ItayBenMoshe) [inline client/src/models/Contact.ts:2]: Why did you remove the '?'

File: server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts [hunk 6]
```diff
@@ -304,4 +290,39 @@ intersectionsRoute.delete('/deleteContactEvent', (request: Request, response: Re
...
+            step: 'got response from DB',
+            investigation: response.locals.epidemiologynumber,
+            user: response.locals.user.id
+        });
+        response.send(result);
+    })
+    .catch(err => {
+        logger.error({
+            service: Service.SERVER,
+            severity: Severity.LOW,
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts:318]: Change to higher severity

File: client/src/models/enums/InteractionsEventDialogContext/InteractionEventContactFields.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@ enum InteractionEventContactFields {
     SERIAL_ID = 'serialId',
     FIRST_NAME = 'firstName',
     LAST_NAME = 'lastName',
-    ID = 'id',
+    ID_NUMBER = 'idNumber',
```
Comment (ItayBenMoshe) [inline client/src/models/enums/InteractionsEventDialogContext/InteractionEventContactFields.ts:5]: Does it not require changes to the db?

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts (hunk 2 above)
Comment (yaniryos) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:79]: does it make a diffrence?

File: client/src/models/enums/InteractionsEventDialogContext/InteractionEventContactFields.ts (hunk 7 above)
Comment (yaniryos) [reply client/src/models/enums/InteractionsEventDialogContext/InteractionEventContactFields.ts:5]: no. only in backend (changed already)

File: client/src/models/Contact.ts (hunk 5 above)
Comment (yaniryos) [reply client/src/models/Contact.ts:2]: because it is always -1 or the value from the DB so it is always defined

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts (hunk 2 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:79]: No, but you should use it.
think that if the field name changes , you will need to make changes only once

File: client/src/models/Contact.ts (hunk 5 above)
Comment (ItayBenMoshe) [reply client/src/models/Contact.ts:2]: ok

Comment (ItayBenMoshe) [reply client/src/models/Contact.ts:2]: So I think you can delete the condition in line 102 in InteractionEventForm.
I added this condition to solve a bug that cause saving multiple contacts every update - make sure it works property

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx [hunk 8]
```diff
@@ -143,32 +145,41 @@ const ContactForm: React.FC<Props> = ({ updatedContactIndex }: Props): JSX.Eleme
...
+        </>
     );
 };
 
 export default ContactForm;
 
 interface Props {
     updatedContactIndex: number;
+    onDeleteClick: () => void;
+    currentItem: any;
```
Comment (roymaniv) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx:184]: למה any

File: server/src/DBService/ContactEvent/Mutation.ts [hunk 9]
```diff
@@ -22,4 +22,12 @@ mutation deleteContactEvent ($contactEventId: Int!) {
     clientMutationId
   }
 }   
+`;
+
+export const DELETE_CONTCTED_PERSON = gql`
```
Comment (roymaniv) [inline server/src/DBService/ContactEvent/Mutation.ts:27]: תשנה ל DELETE_CONTACTED_PERSON
